packages/core/src/Parser.ts

```diff
@@ -222,61 +220,72 @@ export class Parser {
             context.currentNode = node.nextSibling;
             VDocumentMap.set(parsedNode, context.currentNode);
         }
-        this.currentContext = context;
+        return context;
     }
     /**
      * Parse the given text node into VNode(s).
      */
-    _parseTextNode(): void {
+    _parseTextNode(): ParsingContext {
         const node = this.currentContext.currentNode;
         const parentVNode = this.currentContext.parentVNode;
         const format = this.currentContext.format[0];
-        const parsingMap = this.parseNode()[1];
+        const [context, parsingMap] = this.parseNode();
         Array.from(parsingMap.keys()).forEach((parsedVNode: CharNode, index: number) => {
             parsedVNode.format = { ...parsedVNode.format, ...format };
             VDocumentMap.set(parsedVNode, node, index);
             parentVNode.append(parsedVNode);
         });
+        return context;
     }
-    _nextParsingContext(): void {
-        let nextContext = { ...this.currentContext };
-
-        const node = this.currentContext.currentNode;
+    _nextParsingContext(context: ParsingContext): void {
+        const node = context.currentNode;
         if (node.childNodes.length) {
             // Parse the first child with the current node as parent, if any.
-            nextContext.currentNode = node.childNodes[0];
+            context.currentNode = node.childNodes[0];
             // Text node cannot have children, therefore `node` is an Element, not
             // a text node. Only text nodes can be represented by multiple VNodes,
             // so the first matching VNode can safely be selected from the map.
             // If `node` as no VNode representation  in the map (e.g. format nodes),
             // its children are added into the current `parentVNode`.
             const parsedParent = VDocumentMap.fromDom(node);
             if (parsedParent) {
-                nextContext.parentVNode = parsedParent[0];
+                context.parentVNode = parsedParent[0];
             }
-            this._contextStack.push(nextContext);
+            this._contextStack.push(context);
         } else if (node.nextSibling) {
             // Parse the siblings of the current node with the same parent, if any.
-            nextContext.currentNode = node.nextSibling;
+            this.currentContext.currentNode = node.nextSibling;
```

Comment:
Weird... sometimes this function alters `context`, sometimes it alters `this.currentContext`. Red alert, red alert ! 🔴